File: AtronFront/src/app/features/usuarios/models/request/usuario-request.ts
import { UsuarioModel } from "../usuario.model";

export class UsuarioRequest implements UsuarioModel {
  codigo: string;
  nome: string;
  sobrenome: string;
  dataNascimento?: Date | string;
  email: string;
  senha?: string;
  clientUri?: string;
  cargoCodigo: string;
  departamentoCodigo: string;

  constructor(
    codigo: string,
    nome: string,
    sobrenome: string,
    email: string,
    cargoCodigo: string,
    departamentoCodigo: string,
    clientUri?: string,
    dataNascimento?: Date | string) {
    this.codigo = codigo;
    this.nome = nome;
    this.sobrenome = sobrenome;
    this.email = email;
    this.cargoCodigo = cargoCodigo;
    this.departamentoCodigo = departamentoCodigo;
    this.clientUri = clientUri;
    this.dataNascimento = dataNascimento
  }
}

File: AtronTracker/Application/DTO/Request/UsuarioRequest.cs
using System;

namespace Application.DTO.Request
{
    public class UsuarioRequest
    {
        public string Codigo { get; set; }

        public string Nome { get; set; }

        public string Sobrenome { get; set; }

        public DateTime? DataNascimento { get; set; }

        public string CargoCodigo { get; set; }

        public string DepartamentoCodigo { get; set; }

        public string CodigoPerfilDeAcesso { get; set; }

        public string Email { get; set; }

        public string Senha { get; set; }

        public string ClientUri { get; set; }
    }
}
